src/test/scala/edu/ie3/simona/model/em/opt/OptimizedFlexStratSpec.scala

```diff
@@ -0,0 +1,929 @@
+                .currentEnergy
+            )
+
+            // comparing both models:
+            // battery should be exactly empty
+            results(0) should approximate(zeroKWh)
+            results(1) should approximate(zeroKWh)
+
+          }
+        }
+      }
+
+    }
+
+    "provided with simple battery flex options with losses" should {
+
+      given ticks: Seq[Long] =
+        Range.Long.inclusive(0, 4 * sampleTicks, sampleTicks)
+
+      // low efficiency for simplicity of the test
+      val batFlex = EnergyBoundariesFlexOptions(
+        AssetEnergyBoundaries(
+          eStorage = KilowattHours(24),
+          currentEnergy = KilowattHours(12),
+          pMax = Kilowatts(10),
+          etaCharge = Each(0.8),
+          etaDischarge = Each(0.8),
+          currentTick = 0L,
+        )
+      )
+
+      // since energy values have been adapted, we need this
+      // factor to convert back to "real" values
+      given EnergyConversionFactor =
+        EnergyConversionFactor(
+          batFlex.energyBoundaries.headOption.value.etaCharge,
+          OptimizedFlexStrat
+            .adaptEnergyBoundaries(batFlex.energyBoundaries.headOption.value)
+            .etaCharge,
+        )
+
+      "balance out additional power within maximum battery power" in {
+
+        given model: MPModel = MPModel(SolverLib.oJSolver)
+
+        val constFlex = EnergyBoundariesFlexOptions(
+          AssetEnergyBoundaries(
+            Seq(5, -10, 10, -2).toPowerMap
+          )
+        )
+
+        val flexOptions = Map(
+          loadUUID -> constFlex,
+          batUUID -> batFlex,
+        )
+
+        val assetVars =
+          OptimizedFlexStrat.addAssetConstraints(flexOptions, sampleTime, ticks)
+
+        assetVars.toSeq should have length 2
+        assetVars.foreach(_.results should have length 1)
+        assetVars.foreach(_.results.foreach(_ should have length 4))
+
+        val objectiveContainer = OptimizedFlexStrat.buildObjective(
+          assetVars,
+          zeroKW,
+          MinAbsPowerObjectiveFactory,
+        )
+
+        model.minimize(objectiveContainer.objective)
+        model.start(timeLimit = 10000)
+
+        model.getStatus shouldBe SolutionStatus.OPTIMAL
+
+        /*
+          EXPECTED RESULTS
+          Battery should be able to fully cover the additional power
+         */
+
+        val batVars = assetVars
+          .find(_.assetUuid == batUUID)
+          .getOrElse(fail(s"No asset variables for battery ($batUUID) found."))
+
+        val batRes = batVars.results.headOption
+          .getOrElse(fail(s"Empty results for battery ($batUUID)."))
+
+        {
+          objectiveContainer.softConstraints.foreach { constraint =>
+            withClue(constraint.getWarningMessage) {
+              constraint.getError should be < constraintTolerance
+            }
+          }
+
+          // discharging 5 kWh plus 1.25 kWh losses
+          batRes(0).pVal should approximate(-5)
```

Comment:
The switched sign is a bit confusing, maybe it is right because it refers to a change in point of view. However perhaps you see a change to adapt this? (Compared to the Seq you provided to create the AssetEnergyBoundaries)

Reply:
Alright, perhaps it is possible to have the sequence as extra variable named fixedFeedIn, or some comment that this mock a fixed feed-in...